text/0000-integer_atomics.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: integer_atomics
+- Start Date: 2016-03-14
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC basically changes `core::sync::atomic` to look like this:
+
+```rust
+#[cfg(target_has_atomic = "8")]
+struct AtomicBool {}
+#[cfg(target_has_atomic = "8")]
+struct AtomicI8 {}
+#[cfg(target_has_atomic = "8")]
+struct AtomicU8 {}
+#[cfg(target_has_atomic = "16")]
+struct AtomicI16 {}
+#[cfg(target_has_atomic = "16")]
+struct AtomicU16 {}
+#[cfg(target_has_atomic = "32")]
+struct AtomicI32 {}
+#[cfg(target_has_atomic = "32")]
+struct AtomicU32 {}
+#[cfg(target_has_atomic = "64")]
+struct AtomicI64 {}
+#[cfg(target_has_atomic = "64")]
+struct AtomicU64 {}
+#[cfg(target_has_atomic = "128")]
+struct AtomicI128 {}
+#[cfg(target_has_atomic = "128")]
+struct AtomicU128 {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicIsize {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicUsize {}
+#[cfg(target_has_atomic = "ptr")]
+struct AtomicPtr<T> {}
```

Comment:
So I ran a few tests, and here's the current situation for platforms that don't support atomic types (I tested with ARMv6-M):
- LLVM will convert `fetch_add` into a call to ` __sync_fetch_and_add_4`, which is normally provided by compiler-rt.
- However compiler-rt itself uses atomic operations to implement `__sync_fetch_and_add_4`, so it fails to compile on such platforms anyways.

So basically, atomic operations on those platforms has never worked in the first place, so we really aren't breaking anything.